lib/transformers/summary.js

```diff
@@ -15,40 +15,43 @@ module.exports = ({schemaKey, schema, options, data, fullUiSchema} = {}) => {
     } = schema.properties.summaryInfo;
     const answers = answerHelper.summaryFormatter(data, fullUiSchema, lookup);
     let summaryLists = '';
-    summaryStructure.forEach(summaryList => {
-        const rows = [];
-        Object.keys(summaryList.questions).forEach(question => {
-            if (answers[question]) {
-                const row = {
+    summaryStructure.forEach(summarySection => {
+        const questions = [];
+        // For each question in the structure
+        summarySection.questions.forEach(question => {
+            // If the ID exists in the answers
+            if (Object.keys(answers).includes(question.id)) {
+                // Push a summary-shaped nunjucks object into the array
+                questions.push({
                     key: {
-                        text: summaryList.questions[question],
+                        text: question.label,
                         classes: 'govuk-!-width-one-half'
                     },
                     value: {
-                        html: answers[question]
+                        html: answers[question.id]
                     },
                     actions: {
                         items: [
                             {
                                 href: `/${urlPath}/${removeSectionIdPrefix(
-                                    question
+                                    question.id
                                 )}?next=${removeSectionIdPrefix(schemaKey)}`,
                                 text: editAnswerText,
-                                visuallyHiddenText: summaryList.questions[question]
+                                visuallyHiddenText: question.label
                             }
                         ]
                     }
-                };
-
-                rows.push(row);
+                });
             }
+            // Else ignore it.
         });
-
-        if (rows.length > 0) {
-            summaryLists += `<h2 class="govuk-heading-l">${summaryList.title}</h2>
+        // If any questions were found
+        if (questions.length > 0) {
+            // Complete the govukSummaryList object by including a class. Section title also added here.
+            summaryLists += `<h2 class="govuk-heading-l">${summarySection.title}</h2>
```

Comment:
govuk design system updates?

Reply:
Still not included. The `<h2>` isn't a part of the summary, but it part of the 'check your answers' pattern
https://design-system.service.gov.uk/patterns/check-answers/

Reply:
Fair enough!